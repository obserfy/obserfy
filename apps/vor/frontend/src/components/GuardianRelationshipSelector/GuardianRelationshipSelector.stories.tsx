import { Meta, Story } from "@storybook/react"
import GuardianRelationshipSelector, {
  GuardianRelationshipSelectorProps,
} from "./GuardianRelationshipSelector"

export default {
  title: "Core/GuardianRelationshipSelector",
  component: GuardianRelationshipSelector,
  parameters: {
    componentSubtitle: "Just a simple GuardianRelationshipSelector",
  },
} as Meta

const Template: Story<GuardianRelationshipSelectorProps> = (args) => (
  <GuardianRelationshipSelector {...args} />
)

export const Default = Template.bind({})
Default.args = {}
