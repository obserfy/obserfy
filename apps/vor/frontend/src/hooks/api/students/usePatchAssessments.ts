import { useMutation } from "react-query"
import { track } from "../../../analytics"
import { patchApi } from "../fetchApi"
import { useGetStudentMaterialProgressCache } from "../useGetStudentAssessments"

interface RequestBody {
  stage: number
}

const usePatchAssessments = (studentId: string, materialId: string) => {
  const cache = useGetStudentMaterialProgressCache(studentId)
  const patchStudentMaterialProgress = patchApi<RequestBody>(
    `/students/${studentId}/materialsProgress/${materialId}`
  )

  return useMutation(patchStudentMaterialProgress, {
    onSuccess: async (data) => {
      track("Student Material Progress Updated")
      if (!data) return
      const result = await data.json()

      // populate cache if empty
      const oldData = cache.getData()
      if (!oldData) {
        cache.setData([result])
        return
      }

      // if assessments exists in cache, update it
      const newData = [...oldData]
      const idx = newData.findIndex((m) => materialId === m.materialId)
      if (idx >= 0) {
        newData[idx] = result
        cache.setData(newData)
        return
      }

      // if assessments doesn't exists in cache, create it
      newData.push(result)
      cache.setData(newData)
      await cache.invalidate()
    },
  })
}

export default usePatchAssessments
