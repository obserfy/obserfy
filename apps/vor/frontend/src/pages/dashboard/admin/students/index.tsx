import { FC } from "react"
import SEO from "../../../../components/seo"
import PageListOfStudents from "../../../../components/PageListOfStudents/PageListOfStudents"

const Class: FC = () => {
  return (
    <>
      <SEO title="Students" />
      <PageListOfStudents />
    </>
  )
}

export default Class
