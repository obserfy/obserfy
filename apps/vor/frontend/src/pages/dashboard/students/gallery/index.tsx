import PageGallery from "../../../../components/PageGallery/PageGallery"
import { useQueryString } from "../../../../hooks/useQueryString"

const GalleryPage = () => {
  const studentId = useQueryString("studentId")
  return <PageGallery studentId={studentId} />
}

export default GalleryPage
