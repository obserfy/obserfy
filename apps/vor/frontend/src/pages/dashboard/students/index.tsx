import { FC } from "react"
import PageHome from "../../../components/PageHome/PageHome"
import SEO from "../../../components/seo"

const IndexPage: FC = () => {
  return (
    <>
      <SEO title="Home" />
      <PageHome />
    </>
  )
}

export default IndexPage
