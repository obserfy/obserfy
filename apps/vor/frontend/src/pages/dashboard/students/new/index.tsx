import { PageProps } from "gatsby"
import { FC } from "react"
import PageNewStudent from "../../../../components/PageNewStudent/PageNewStudent"
import SEO from "../../../../components/seo"

const NewStudent: FC<PageProps> = () => {
  return (
    <>
      <SEO title="Edit Student" />
      <PageNewStudent />
    </>
  )
}
export default NewStudent
