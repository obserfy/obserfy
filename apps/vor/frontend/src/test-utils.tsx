/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
