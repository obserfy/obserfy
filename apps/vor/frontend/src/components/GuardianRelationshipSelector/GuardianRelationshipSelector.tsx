import { t } from "@lingui/macro"
import { useLingui } from "@lingui/react"
import { ChangeEventHandler, FC } from "react"
import { Box } from "theme-ui"
import { GuardianRelationship } from "../../hooks/api/students/usePostNewStudent"
import Select from "../Select/Select"

export interface GuardianRelationshipSelectorProps {
  value: GuardianRelationship
  onChange: ChangeEventHandler<HTMLSelectElement>
}
const GuardianRelationshipSelector: FC<GuardianRelationshipSelectorProps> = ({
  onChange,
  value,
}) => {
  const { i18n } = useLingui()

  return (
    <Box px={3} pt={3}>
      <Select label={t`Relationship`} value={value} onChange={onChange}>
        <option value={GuardianRelationship.Other}>{i18n._(t`Other`)}</option>
        <option value={GuardianRelationship.Mother}>{i18n._(t`Mother`)}</option>
        <option value={GuardianRelationship.Father}>{i18n._(t`Father`)}</option>
      </Select>
    </Box>
  )
}

export default GuardianRelationshipSelector
