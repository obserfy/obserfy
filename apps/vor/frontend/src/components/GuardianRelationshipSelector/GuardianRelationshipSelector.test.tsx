import { GuardianRelationship } from "../../hooks/api/students/usePostNewStudent"
import { render } from "../../test-utils"
import GuardianRelationshipSelector from "./GuardianRelationshipSelector"

describe("GuardianRelationshipSelector", () => {
  it("should render correctly", () => {
    const { container } = render(
      <GuardianRelationshipSelector
        onChange={() => {}}
        value={GuardianRelationship.Mother}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
