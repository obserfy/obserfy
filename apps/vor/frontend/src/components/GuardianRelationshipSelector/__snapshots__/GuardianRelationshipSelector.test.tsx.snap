// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuardianRelationshipSelector should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-bottom: 4px;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: var(--theme-ui-colors-background);
  border-radius: 6px;
  color: var(--theme-ui-colors-text);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
}

.emotion-5:hover,
.emotion-5:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -28px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  pointer-events: none;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <label
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          Relationship
        </div>
        <div
          class="emotion-4"
        >
          <select
            aria-label="Relationship"
            class="emotion-5"
          >
            <option
              value="0"
            >
              Other
            </option>
            <option
              value="1"
            >
              Mother
            </option>
            <option
              value="2"
            >
              Father
            </option>
          </select>
          <svg
            class="emotion-6"
            fill="currentcolor"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </label>
    </div>
  </div>
</div>
`;
